From cd1159c8b3a25591dd633c76fd1547d69b17ef61 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 29 Oct 2014 16:34:19 -0400
Subject: [fs] ext4: fix accidental flag aliasing in ext4_map_blocks flags

Message-id: <1414600485-1929-30-git-send-email-lczerner@redhat.com>
Patchwork-id: 98551
O-Subject: [RHEL 7.1 PATCH v2 29/55] ext4: fix accidental flag aliasing in ext4_map_blocks flags
Bugzilla: 1156096
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1156096
https://bugzilla.redhat.com/show_bug.cgi?id=1156096

BREW 8169722
https://brewweb.devel.redhat.com/taskinfo?taskID=8169722

Upstream commit bd30d702fc320085f178d22866b32fdc4736c991

 Commit b8a8684502a0f introduced an accidental flag aliasing between
 EXT4_EX_NOCACHE and EXT4_GET_BLOCKS_CONVERT_UNWRITTEN.

 Fortunately, this didn't introduce any untorward side effects --- we
 got lucky.  Nevertheless, fix this and leave a warning to hopefully
 avoid this from happening in the future.

 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/ext4.h b/fs/ext4/ext4.h
index 512579d..ffcde03 100644
--- a/fs/ext4/ext4.h
+++ b/fs/ext4/ext4.h
@@ -572,6 +572,7 @@ enum {
 #define EXT4_GET_BLOCKS_NO_PUT_HOLE  0x0200
  /* Convert written extents to unwritten */
 #define EXT4_GET_BLOCKS_CONVERT_UNWRITTEN 0x0400
+/* DO NOT ASSIGN ADDITIONAL FLAG VALUES WITHOUT ADJUSTING THE FLAGS BELOW */
 
 /*
  * The bit position of these flags must not overlap with any of the
@@ -582,8 +583,8 @@ enum {
  * caching the extents when reading from the extent tree while a
  * truncate or punch hole operation is in progress.
  */
-#define EXT4_EX_NOCACHE    0x0400
-#define EXT4_EX_FORCE_CACHE   0x0800
+#define EXT4_EX_NOCACHE    0x0800
+#define EXT4_EX_FORCE_CACHE   0x1000
 
 /*
  * Flags used by ext4_free_blocks
-- 
1.7.1